crates/router/src/core/admin.rs

```diff
@@ -2001,3 +2040,309 @@ pub fn validate_status_and_disabled(
 
     Ok((connector_status, disabled))
 }
+
+async fn process_open_banking_connectors(
+    state: &AppState,
+    merchant_id: &str,
+    key_store: &domain::MerchantKeyStore,
+    auth: &types::ConnectorAuthType,
+    connector_type: &api_enums::ConnectorType,
+    connector: &api_enums::Connector,
+    merchant_data: Secret<serde_json::Value>,
+) -> RouterResult<types::MerchantRecipientData> {
+    // incorporate a connector check as well
+    if connector_type != &api_enums::ConnectorType::PaymentProcessor {
+        return Err(errors::ApiErrorResponse::InvalidConnectorConfiguration {
+            config: "OpenBankingAuth type needs to be coupled with a payment connector".to_string(),
+        }
+        .into());
+    }
+
+    let additional_merchant_data =
+        serde_json::from_value::<types::AdditionalMerchantData>(merchant_data.expose())
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("failed to decode MerchantRecipientData")?;
+
+    let new_merchant_data = match additional_merchant_data {
+        types::AdditionalMerchantData::OpenBankingRecipientData(merchant_data) => {
+            match &merchant_data {
+                types::MerchantRecipientData::AccountData(acc_data) => {
+                    validate_bank_account_data(acc_data)?;
+
+                    let connector_name = api_enums::Connector::to_string(connector);
+
+                    let locker_based_connector_list =
```

Comment:
Cloning is not required here.

Reply:
The config is inside an `Arc` so cloning would be required right?